bioptim/models/biorbd/biorbd_model.py

```diff
@@ -276,165 +425,245 @@ def muscle_names(self) -> tuple[str, ...]:
+                    name = self._segments_to_apply_forces_in_global[i_element]
+                    values = forces_in_global[:6, i_element]
+                    point_of_application = forces_in_global[6:9, i_element]
+                    external_forces_set.add(name, values, point_of_application)
+
+            if forces_in_local.shape[1] > 0:
+                # Add the external forces in the local reference frame
+                for i_element in range(forces_in_local.shape[1]):
+                    name = self._segments_to_apply_forces_in_local[i_element]
+                    values = forces_in_local[:6, i_element]
+                    point_of_application = forces_in_local[6:9, i_element]
+                    external_forces_set.addInSegmentReferenceFrame(name, values, point_of_application)
+
+            elif translational_forces.shape[1] > 0:
+                # Add the translational forces
+                for i_elements in range(translational_forces.shape[1]):
+                    name = self._segments_to_apply_translational_forces[i_elements]
+                    values = translational_forces[:3, i_elements]
+                    point_of_application = translational_forces[3:6, i_elements]
+                    external_forces_set.addTranslationalForce(values, name, point_of_application)
+
+            return external_forces_set
+
+    def forward_dynamics(self, with_contact: bool = False) -> Function:
+
+        q_biorbd = GeneralizedCoordinates(self.q)
+        qdot_biorbd = GeneralizedVelocity(self.qdot)
+        tau_biorbd = GeneralizedTorque(self.tau)
+
+        if with_contact:
+            if self.external_forces_set is None:
+                biorbd_return = self.model.ForwardDynamicsConstraintsDirect(q_biorbd, qdot_biorbd, tau_biorbd).to_mx()
+            else:
+                biorbd_return = self.model.ForwardDynamicsConstraintsDirect(
+                    q_biorbd, qdot_biorbd, tau_biorbd, self.external_forces_set
+                ).to_mx()
+            casadi_fun = Function(
+                "constrained_forward_dynamics",
+                [self.q, self.qdot, self.tau, self.external_forces, self.parameters],
+                [biorbd_return],
+                ["q", "qdot", "tau", "external_forces", "parameters"],
+                ["qddot"],
+            )
+        else:
+            if self.external_forces_set is None:
+                biorbd_return = self.model.ForwardDynamics(q_biorbd, qdot_biorbd, tau_biorbd).to_mx()
+            else:
+                biorbd_return = self.model.ForwardDynamics(
+                    q_biorbd, qdot_biorbd, tau_biorbd, self.external_forces_set
+                ).to_mx()
+            casadi_fun = Function(
+                "forward_dynamics",
+                [self.q, self.qdot, self.tau, self.external_forces, self.parameters],
+                [biorbd_return],
+                ["q", "qdot", "tau", "external_forces", "parameters"],
+                ["qddot"],
+            )
```

Comment:
If possible otherwise, we can forget it.
```python3
        dynamics_functions = {
            (True, True): nlp.model.constrained_forward_dynamics,
            (True, False): nlp.model.forward_dynamics,
            (False, True): lambda q, qdot, tau: nlp.model.constrained_forward_dynamics(q, qdot, tau, external_forces),
            (False, False): lambda q, qdot, tau: nlp.model.forward_dynamics(q, qdot, tau, external_forces),
        }

        dynamics_function = dynamics_functions[(external_forces is None, with_contact)]

        qddot = dynamics_function(q, qdot, tau)
        return qdot_var_mapping.map(qddot)
        
```